db/update-files.d/014_alter_lubw_parser.sql

```diff
@@ -0,0 +1,249 @@
+\c sauber_data
+
+/*
+Changes: Check that input coordinates not null 
+*/
+
+CREATE OR REPLACE FUNCTION station_data.prediction_parse()
+ RETURNS text
+ LANGUAGE plpgsql
+AS $function$
+DECLARE
+  payload JSONB;
+  message JSONB;
+  prediction JSONB;
+  j JSONB;
+  pred_dt TIMESTAMP;
+  pred_val DOUBLE PRECISION;
+
+  message_timestamp TIMESTAMP;
+  category_name TEXT;
+  message_payload JSONB;
+  
+  component_id TEXT;
+  component_id_short TEXT;
+  unit_id TEXT;
+  region_id TEXT;
+  station_id TEXT;
+  coordinates_text TEXT;
+  coordinates GEOMETRY;
+  interval_len INT;
+  creation_time TIMESTAMP;
+  prediction_end_time TIMESTAMP;
+  prediction_start_time TIMESTAMP;
+
+  time_to_prediction SMALLINT;
+
+  logentry_payload JSONB;
+  counter INTEGER;
+BEGIN
+
+  DROP TABLE IF EXISTS tmp_json_vals;
+
+  -- Create tmp tables to hold parsed values
+  CREATE TEMP TABLE tmp_json_vals (
+    tmp_dt TIMESTAMP,
+    tmp_val DOUBLE PRECISION,
+    tmp_station TEXT,
+    tmp_geom GEOMETRY,
+    tmp_region TEXT,
+    tmp_component TEXT,
+    tmp_timetopred SMALLINT
+  ) ON COMMIT DROP;
+
+  -- Get last raw JSON FROM  input table
+  -- This function gets called only after successful JSON insert
+  --> Last JSON should be latest input
+  SELECT json_payload
+    FROM  station_data.input_prediction
+    ORDER BY idpk_json DESC
+    LIMIT 1
+  INTO payload;
+
+
+  SELECT json_message
+    FROM  station_data.input_prediction
+    ORDER BY idpk_json DESC
+    LIMIT 1
+  INTO message;
+
+
+  RAISE NOTICE '%', message;
+
+  -- Read message parameters
+
+  message_timestamp := to_timestamp((message->'timestamp')::bigint);
+  category_name := message->'category';
+  message_payload := message->'payload';
+
+  component_id := message_payload->>'type';
+  component_id_short := split_part(message_payload->>'type','_',1);
+  unit_id := message_payload->>'unit';
+  region_id := message_payload->>'region';
+  interval_len := message_payload->'interval';
+  station_id := message_payload->>'stationId';
+  coordinates_text := message_payload->>'coordinates';	 
+  creation_time := to_timestamp((message_payload->'creationTime')::bigint);
+  prediction_end_time := to_timestamp((message_payload->'predictionEndTime')::bigint);
+  prediction_start_time := to_timestamp((message_payload->'predictionStartTime')::bigint);
+  prediction := payload->'prediction';
+
+  -- Check if coords are empty
+  -- If empty -> Assign coordinates null
+	 IF (coordinates_text = '') IS TRUE THEN 
+	   coordinates := NULL;
+	 ELSE 
+	   coordinates := coordinates_text::GEOMETRY;
```

Comment:
looks good

spontaneous idea:
maybe be could add [`ST_isValid()`](https://postgis.net/docs/ST_IsValid.html) to catch any other errors of the input data ...